### PR TITLE
 Adding update-gofmt.sh to fix gofmt errors 

### PR DESCRIPTION
**Pull request by poothia**
Adding update-gofmt.sh to automate fixing of gofmt errors.
part of #263 

/sig multicluster

### CODE REVIEW

File: hack/go-tools/update-gofmt.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eou pipefail
+
+find . -name "*.go" | grep -v "\/vendor\/" | xargs gofmt -s -w
```
Comment (font) [inline hack/go-tools/update-gofmt.sh]: This should support running against the root of the repo regardless from where the script is executed i.e. `.` is insufficient here. See the other scripts for an example such as `verify-gofmt.sh`.

**Comment by poothia** [comment]
Thanks for reviewing this, I have addressed your comment.

File: hack/go-tools/update-gofmt.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright 2017 The Kubernetes Authors.
```
Comment (font) [inline hack/go-tools/update-gofmt.sh]: 2018

File: hack/go-tools/update-gofmt.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eou pipefail
+
+SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+REPO_ROOT="$(cd "${SCRIPT_ROOT}/../.." && pwd)"
+pushd ${REPO_ROOT} > /dev/null
+
+find . -name "*.go" | grep -v "\/vendor\/" | xargs gofmt -s -w
```
Comment (font) [inline hack/go-tools/update-gofmt.sh]: Although unnecessary, please add a `popd` just for completion.

**Comment by poothia** [comment]
done.

**Comment by gyliu513** [comment]
@font Does the CI has some issue? Seems the CI not getting `vendor` source code and caused the build failed.

My two PRs failing because of same reason
- https://github.com/kubernetes/cluster-registry/pull/259
- https://github.com/kubernetes/cluster-registry/pull/258

**Comment by poothia** [comment]
@font  @gyliu513  is there anything else that should be done on this PR?

**Comment by gyliu513** [comment]
@font can we get this merged? Thanks.

**Comment by poothia** [comment]
@font PTAL.

**Pull request closed without merging**